api/v1alpha1/schedule_types.go

```diff
@@ -19,10 +19,13 @@ type ScheduleSpec struct {
 	ResourceRequirementsTemplate corev1.ResourceRequirements `json:"resourceRequirementsTemplate,omitempty"`
 }
 
+// ScheduleDefinition is the actual cron-type expression that defines the interval of the actions.
+type ScheduleDefinition string
```

Comment:
This change was not strictly necessary. It's a leftover from me trying to improve the critiqued 'if-cascade'. I figured it improved the code and therefore left this change in place.

Reply:
If we're making a type for a string, at least we should add a "String()" method, that's a bit easier to use and read than having to use string(...)

Reply:
Fair enough!

Reply:
Although it isn't actually needed as I just figured out, as this simple example demonstrates:

```go
package main

import (
	"fmt"
)

type Foo string

func main() {
	foo := Foo("Hello, playground")
	fmt.Println(foo)
}
```

(Try it on https://play.golang.org/)

But there are other interesting things that can be done when that silly string is strongly typed. I've implemented two changes: _func (s ScheduleDefinition) IsRegularCron() bool_ and _func (s ScheduleDefinition) IsRandom() bool_. With it, IMO the flow of `getEffectiveSchedule` in `schedule.go` is easier to comprehend.